Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityTeleport.cs

```diff
@@ -27,16 +27,28 @@ namespace HoloToolkit.Unity.InputModule
     public class MixedRealityTeleport : Singleton<MixedRealityTeleport>, IControllerInputHandler
     {
         [Tooltip("Name of the thumbstick axis to check for teleport and strafe.")]
-        public string LeftThumbstickX = "CONTROLLER_LEFT_STICK_HORIZONTAL";
+        public XboxControllerMappingTypes HorizontalStrafe = XboxControllerMappingTypes.XboxLeftStickHorizontal;
 
         [Tooltip("Name of the thumbstick axis to check for teleport and strafe.")]
-        public string LeftThumbstickY = "CONTROLLER_LEFT_STICK_VERTICAL";
+        public XboxControllerMappingTypes VerticalStrafe = XboxControllerMappingTypes.XboxLeftStickVertical;
 
         [Tooltip("Name of the thumbstick axis to check for rotation.")]
-        public string RightThumbstickX = "CONTROLLER_RIGHT_STICK_HORIZONTAL";
+        public XboxControllerMappingTypes HorizontalRotation = XboxControllerMappingTypes.XboxRightStickHorizontal;
 
         [Tooltip("Name of the thumbstick axis to check for rotation.")]
-        public string RightThumbstickY = "CONTROLLER_RIGHT_STICK_VERTICAL";
+        public XboxControllerMappingTypes VerticalRotation = XboxControllerMappingTypes.XboxRightStickVertical;
```

Comment:
We don't have vertical rotation set up. The right stick's Y is used for intent checking when the right stick's X is within range to trigger a rotation, that it's mostly in a right/left direction.

Reply:
If it wasn't setup then why was it mapped?

Should we remove it/implement it?

Reply:
> The right stick's Y is used for intent checking when the right stick's X is within range to trigger a rotation, that it's mostly in a right/left direction.

It's used to make sure the stick isn't too high or low when pressed to the left or right, to try to filter out false positives.

https://github.com/StephenHodgson/MixedRealityToolkit-Unity/blob/49133ed42330ec96674caaa7e5ce6a8fc2901c6e/Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityTeleport.cs#L171

Reply:
So what should the proper name be?

Reply:
Good question. I'm not completely convinced the drop-downs here will work, since they're used for more than just one thing.

Thumbstick X is used for both teleporting forward and preventing false positives on the Thumbstick Y strafing.

Thumbstick Y is used for horizontal strafing and preventing false positives on the Thumbstick X teleporting forward. 

If I set the new Horizontal Strafe to `Xbox Left Stick Horizontal` and Forward Movement to `Xbox Dpad Vertical`, the false positive detection becomes useless.
Of course, maybe we don't really need it! If the horizontal strafe values are set to values that physically can't be returned once the thumbstick is at a certain vertical value, those checks aren't needed anyway.

Additionally, if I map `Horizontal Strafe` to something that isn't a (-1,1) axis, like `Xbox A`, it strafes me to the right on every press, with no way to go left, but hopefully that's something the dev will be able to figure out isn't supported.

Hmm, also using `Xbox Dpad Vertical` in my scenario above would break teleporting with motion controllers, since they don't have a Dpad. That's why we defaulted to only using thumbsticks (and that it matches the shell).

One last thought I had is that all of these options start with "Xbox", which might be misleading, since they're used for the motion controller teleporting as well.

Reply:
I thought about renaming the Enum but I didn't want to introduce another breaking change.

Reply:
Essentially I'm trying to add customization to the component.

Reply:
Do you think these values should be hard coded?

Reply:
Hmm, good point. In this case, we tailored the experience to match the shell, but it's possible devs would want to use buttons or other inputs instead of the thumbstick. I'm not sure they'd want to use different values though, since they're basically "is the thumbstick pushed forward/left/right/etc".

Reply:
I think any mapped input that is an Axis should be valid for this type of input.